modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Editor} from './Editor';
+
+import '../css/main.scss';
+
+const BalloonEditor = ({config = {}, contents, name, ...otherProps}) => {
+	const defaultExtraPlugins = 'balloontoolbar,floatingspace';
+
+	const getConfig = () => {
+		const extraPlugins = config.extraPlugins
+			? `${config.extraPlugins}`
+			: '';
+
+		return {
+			...config,
+			extraAllowedContent: '*',
+			extraPlugins: `${extraPlugins}${defaultExtraPlugins}`,
+		};
+	};
+
+	let initialCssClass;
```

Comment:
Just to note a potential point of fragility here. _If_ this component ever re-rerenders, and _if_ it doesn't reliably trigger `onBeforeLoad` and `onDestroy` in perfect matching pairs, it is possible that the value that gets restored on line 51 will not be the one that got saved on line 46...

And also, that looks like a global variable... what's going to happen if there are multiple editors on the page fighting to save/restore the "initial" class? 😱 (Maybe this was discussed previously, so please forgive me if I missed that.)

Reply:
> And also, that looks like a global variable...
Yes, once more thing I didn't think about correctly

> what's going to happen if there are multiple editors on the page fighting to save/restore the "initial" class?
Probably something I don't want :joy:

> (Maybe this was discussed previously, so please forgive me if I missed that.)

It hasn't been discussed so thanks for the heads up. 
I'm thinking this should be in the component's "state" (i.e. `useState`)